src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -153,15 +160,21 @@ GenericNameSyntax syntax
 					continue;
 				}
 
+				// If the syntax is a GenericName (has explicit type arguments) then the error should be on the argument
+				// Otherwise, it should be on the identifier itself

```

Comment:
Thanks for the comment. Maybe it should go beside the `getLocation:` argument though?